lib/Parse/ParseGeneric.cpp

```diff
@@ -368,6 +368,13 @@ ParserStatus Parser::parseGenericWhereClause(
     BodyContext.reset();
     HasNextReq = consumeIf(tok::comma);
     // If there's a comma, keep parsing the list.
+    // If thers's a "&&", diagnose replace with comma and keep parsing
+    if (Tok.isBinaryOperator() && Tok.getText() == "&&") {
```

Comment:
This is also going to parse `One , && Two`.  You should add a check for `!HasNextReq` to this condition.